dotnet/src/webdriver/PinnedScript.cs

```diff
@@ -17,83 +17,70 @@
         /// <remarks>
         /// This constructor is explicitly internal. Creation of pinned script objects
         /// is strictly the perview of Selenium, and should not be required by external
         /// libraries.
         /// </remarks>
-        internal PinnedScript(string script)
+        internal PinnedScript(string script, string stringHandle, string scriptId)
         {
-            this.scriptSource = script;
-            this.scriptHandle = Guid.NewGuid().ToString("N");
+            this.Source = script;
+            this.Handle = stringHandle;
+            this.ScriptId = scriptId;
         }
 
         /// <summary>
         /// Gets the unique handle for this pinned script.
         /// </summary>
-        public string Handle
-        {
-            get { return this.scriptHandle; }
-        }
+        public string Handle { get; }
 
         /// <summary>
         /// Gets the source representing the body of the function in the pinned script.
         /// </summary>
-        public string Source
-        {
-            get { return this.scriptSource; }
-        }
+        public string Source { get; }
 
-        /// <summary>
-        /// Gets the script to create the pinned script in the browser.
-        /// </summary>
-        internal string CreationScript
+        internal static string MakeCreationScript(string scriptHandle, string scriptSource)
         {
-            get { return string.Format(CultureInfo.InvariantCulture, "function __webdriver_{0}(arguments) {{ {1} }}", this.scriptHandle, this.scriptSource); }
+            return string.Format(CultureInfo.InvariantCulture, "function __webdriver_{0}(arguments) {{ {1} }}", scriptHandle, scriptSource);
         }
 
         /// <summary>
         /// Gets the script used to execute the pinned script in the browser.
         /// </summary>
-        internal string ExecutionScript
+        internal string MakeExecutionScript()
         {
-            get { return string.Format(CultureInfo.InvariantCulture, "return __webdriver_{0}(arguments)", this.scriptHandle); }
+            return string.Format(CultureInfo.InvariantCulture, "return __webdriver_{0}(arguments)", this.Handle);
         }
 
         /// <summary>
         /// Gets the script used to remove the pinned script from the browser.
         /// </summary>
-        internal string RemovalScript
+        internal string MakeRemovalScript()
         {
-            get { return string.Format(CultureInfo.InvariantCulture, "__webdriver_{0} = undefined", this.scriptHandle); }
+            return string.Format(CultureInfo.InvariantCulture, "__webdriver_{0} = undefined", this.Handle);
         }
 
         /// <summary>
         /// Gets or sets the ID of this script.
         /// </summary>
-        internal string ScriptId
-        {
-            get { return this.scriptId; }
-            set { this.scriptId = value; }
-        }
+        internal string ScriptId { get; }
```

Comment:
Should be a fild?

Reply:
This value is used by another type, in the `JavaScriptEngine.UnpinScript` method. I believe fields should only be `private`, so we will need some property to expose it.

Reply:
All right.